pam/internal/adapter/qrcodemodel.go

```diff
@@ -1,23 +1,30 @@
 package adapter
 
 import (
+	"context"
 	"fmt"
 	"os"
 	"strings"
+	"time"
 
 	tea "github.com/charmbracelet/bubbletea"
 	"github.com/charmbracelet/lipgloss"
 	"github.com/muesli/termenv"
 	"github.com/skip2/go-qrcode"
 	"github.com/ubuntu/authd"
+	"github.com/ubuntu/authd/internal/log"
 )
 
 var centeredStyle = lipgloss.NewStyle().Align(lipgloss.Center, lipgloss.Top)
 
+// reselectionRateLimit is the amount of time (in ms) that we wait before regenerating the qrcode.
+const reselectionRateLimit = int64(300)
+
 // qrcodeModel is the form layout type to allow authenticating and return a challenge.
 type qrcodeModel struct {
-	label       string
-	buttonModel *buttonModel
+	label              string
+	buttonModel        *buttonModel
+	selectionTimestamp int64
```

Comment:
This should be of type `time.Time`. This way, we don't need to do the conversion all the time, just when we need it and we also gain access to some implementations like `.After()` and so on.